src/lib/expert_partitioner/dialogs/format.rb

```diff
@@ -90,30 +80,32 @@ def create_dialog
       )
     end
 
+    def filesystem_items
+      [
+        Item(Id(Storage::FsType_EXT4), "Ext4"),
+        Item(Id(Storage::FsType_XFS), "XFS"),
+        Item(Id(Storage::FsType_BTRFS), "Btrfs"),
+        Item(Id(Storage::FsType_SWAP), "Swap"),
+        Item(Id(Storage::FsType_NTFS), "NTFS"),
+        Item(Id(Storage::FsType_VFAT), "VFAT")
+      ]
+    end
 
     def doit
-      begin
-
-        log.info "doit #{@blk_device.name}"
+      log.info "doit #{@blk_device.name}"
 
-        if !RemoveDescendantsPopup.new(@blk_device).run()
-          return
-        end
+      return if !RemoveDescendantsPopup.new(@blk_device).run
 
-        filesystem = @blk_device.create_filesystem(Yast::UI.QueryWidget(:filesystem, :Value))
+      filesystem = @blk_device.create_filesystem(Yast::UI.QueryWidget(:filesystem, :Value))
 
-        mount_point = Yast::UI.QueryWidget(:mount_point, :Value)
-        if !mount_point.empty?
-          log.info "doit mount-point #{mount_point}"
-          filesystem.add_mountpoint(mount_point)
-        end
-
-      rescue Storage::DeviceHasWrongType
-        log.error "doit on non blk device"
+      mount_point = Yast::UI.QueryWidget(:mount_point, :Value)
+      if !mount_point.empty?
+        log.info "doit mount-point #{mount_point}"
+        filesystem.add_mountpoint(mount_point)
       end
 
+    rescue Storage::DeviceHasWrongType
+      log.error "doit on non blk device"
```

Comment:
Calling the format dialog for a non-block-device is plain wrong so the exception here is the right thing to do.